src/lib/storage/BufferTreeFs.ts

```
@@ -0,0 +1,43 @@
+import { Member } from './Container'
+import * as fs from 'fs'
+import * as path from 'path'
+
+export interface ReadableResource {
+  body: ReadableStream<Buffer>
+  metaData: { [i: string]: Buffer }
+}
+
+export interface BufferTree {
+  getMembers (url: URL): Promise<Array<Member> | undefined>
+  getResource (url: URL): Promise<ReadableResource>
+  setResource (url: URL, data: ReadableResource): Promise<void>
```

This should be a method on Resource, and should throw an Error in case the resource on disk changed since that Resource object was created (like we do with `watch` in the Redis one)